auth/src/main/java/com/firebase/ui/auth/FirebaseAuthUIActivityResultContract.java

```diff
@@ -0,0 +1,31 @@
+package com.firebase.ui.auth;
+
+import android.content.Context;
+import android.content.Intent;
+
+import com.firebase.ui.auth.data.model.FirebaseAuthUIAuthenticationResult;
+
+import androidx.activity.result.contract.ActivityResultContract;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+/**
+ * A {@link ActivityResultContract} describing that the caller can launch authentication with a
+ * {@link com.firebase.ui.auth.AuthUI.SignInIntentBuilder} and is guaranteed to receive a
+ * {@link FirebaseAuthUIAuthenticationResult} as result.
+ */
+public class FirebaseAuthUIActivityResultContract extends
+        ActivityResultContract<AuthUI.SignInIntentBuilder, FirebaseAuthUIAuthenticationResult> {
+
+    @NonNull
+    @Override
+    public Intent createIntent(@NonNull Context context, AuthUI.SignInIntentBuilder input) {
+        return input.build();
```

Comment:
I'm curious about why we're passing the `IntentBuilder` to the `Contract` instead of the actual `Intent`?

Reply:
Thanks for highlighting this, I wanted to come back to it! I actually don't really like this as input type here. It would be great to use an immutable data structure (`FlowParams`?) instead. I prefer that over an intent because it represents the input more accurately and is more intuitive. Say you're looking at how to use this the first time, an intent can contain anything so it's not necessarily clear how to construct it/what it should contain.

What are your thoughts on that? Do you have any model in mind that we could re-use as input type here? Thanks for looking at it!:)

Reply:
Thanks @rosariopfernandes for flagging this! @jossiwolf I totally get where you're coming from, `Intent` allows too much room for error.

Unfortunately right now we don't have a good internal type to cleanly represent the state of the `IntentBuilder` ... there was never much use for that before this PR.

I think for `v7.2.0` we should take `Intent` as the param here but then in `v8.0.0` we can look at refactoring this to take an immutable input type representing all of the options.

Reply:
Cool, sounds good! This would be great to have in any case I think (I can see this coming in handy for Compose support) so that work could be done separately too.

I'll refactor this to accept an Intent as input, in that case the samples change a bit less too which is great!